client/ayon_core/tools/publisher/control.py

```diff
@@ -1893,9 +1893,9 @@ def _get_allowed_creator_labels(self):
 
     def _label_matches_allowed(self, label, allowed_labels):
         """Implement regex support for allowed labels."""
-        for allowed in allowed_labels:
-            if re.match(allowed, label):
-                return True
+        allowed_patterns = re.compile("|".join(allowed_labels))
```

Comment:
Even better if we can take this join and compile outside of the `for loop` where this method is called.
Which has the benefit that this just turns into:
```python
return bool(pattern.match(label))
```